src/components/ViewRequest/HiringPanel.tsx

```diff
@@ -7,15 +8,20 @@ import { useParams } from "react-router-dom";
 
 const HiringPanel = () => {
   const params = useParams();
-  const request = useRequest(Number(params.requestId));
+  const requestId = Number(params.requestId);
+  const request = useRequest(requestId);
   const postRequest = usePostRequest();
+  const addNote = useAddNote(requestId);
 
   const onChange = useCallback(
     (ev: ChangeEvent<HTMLInputElement>) => {
       postRequest.mutate({
-        requestId: Number(params.requestId),
+        requestId: requestId,
         postRequest: { panelRequired: ev.currentTarget.checked ? "Yes" : "No" },
       });
+      addNote.mutate(
+        `Set 'Panel Required' to ${ev.currentTarget.checked ? "Yes" : "No"}`
+      );
     },
     [params.requestId]
```

Comment:
This dependency is now just "requestId".  However, I think it is a new object on every re-render, so essentially the useCallback is going to be re-generated on every re-render too -- so it wouldn't really be gaining any optimization right?  Unless you switch back to using params.requestId in the mutate call, then it would only change if the param changed.  Also should update to fix the lint warnings for dependencies of the 2 mutate objects.  Couple of options -- see https://github.com/TanStack/query/issues/1858 where TkDodo explains one way -- another user also suggests adding the obj.mutate to the array and using a comment to tell the linter to ignore the warning, but also a comment to explain to us humans why.